lms/djangoapps/grades/tests/test_signals.py

```diff
@@ -148,3 +160,65 @@ def test_score_reset_bad_user(self):
         self.get_user_mock = self.setup_patch('lms.djangoapps.grades.signals.user_by_anonymous_id', None)
         submissions_score_reset_handler(None, **SUBMISSION_RESET_KWARGS)
         self.signal_mock.assert_not_called()
+
+
+@patch.dict('django.conf.settings.FEATURES', {'ENABLE_SUBSECTION_GRADES_SAVED': True})
+@ddt.ddt
+class ScoreChangedUpdatesSubsectionGradeTest(ModuleStoreTestCase):
+    """
+    Ensures that upon SCORE_CHANGED signals, the handler
+    initiates an update to the affected subsection grade.
+    """
+    def setUp(self):
+        """
+        Configure mocks for all the dependencies of the render method
+        """
+        super(ScoreChangedUpdatesSubsectionGradeTest, self).setUp()
+
+        # mock student
+        self.user_mock = MagicMock()
+        self.user_mock.id = 42
+        self.get_user_mock = self.setup_patch('lms.djangoapps.grades.signals.user_by_anonymous_id', self.user_mock)
+        self.course = CourseFactory.create(
+            org='edx',
+            name='course',
+            run='run'
+        )
+        self.course.save()
+        self.store.update_item(self.course, 0)
+        self.chapter = ItemFactory.create(parent=self.course, category="chapter", display_name="Chapter")
+        self.sequential = ItemFactory.create(parent=self.chapter, category='sequential', display_name="Open Sequential")
+        self.problem = ItemFactory.create(parent=self.sequential, category='problem', display_name='problem')
+
+        self.score_changed_kwargs = {
+            'points_possible': 10,
+            'points_earned': 5,
+            'user_id': 'anonymous_id',
+            'course_id': unicode(self.course.id),
+            'usage_id': unicode(self.problem.location)
+        }
+
+        self.update_mock = self.setup_patch(subsection_grade.__name__ + '.SubsectionGradeFactory.update', None)
```

Comment:
I would expect the sql query count to be identical.
The mongo query count may _(or may not)_ differ.
However, did you verify that the course is a split-course when `ModuleStoreEnum.Type.split` is selected?  The course-key should look like a split-course key with `course-v1` and pluses.
Most likely it's not since the `self.store.default_store` context manager is used only after the course is created.  You'll need to refactor your test code so that the course-creation happens within that context manager.